### PR TITLE
[Quantization] Add bias quant

### PR DESCRIPTION
**Pull request by linbinskn**
The target of this pr is to be compatible with int8 kernel implementation, namely int32 biasadd for cpu and int8 biasadd for gpu.

### CODE REVIEW

**Comment by chenbohua3** [comment]
I think it is better to construct a stand-alone pdll file to support bias with fake-quant, since it is not a common way.

**Comment by linbinskn** [comment]
> I think it is better to construct a stand-alone pdll file to support bias with fake-quant, since it is not a common way.

Only add gpu bias quant this pr. Final solution will be discussed later.

File: pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py [hunk 1]
```diff
@@ -63,7 +63,8 @@ def forward(self, x):
                     self.activation_quant_min, self.activation_quant_max
                 )
                 return x
-
+        if self.device == torch.device('cuda'):
```
Comment (chenbohua3) [inline pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py:66]: It is better to build different base classes for different device (e.g. gpu and cpu) to enable testing on the respective hardware. Also, skip test using `return` may not be a good idea, use the api of `unitest` instead

**Comment by qiuxiafei** [comment]
Just adding tests?

**Comment by linbinskn** [comment]
> Just adding tests?

In addition to add test, it also modifies and adds pdll files.

File: pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py (hunk 1 above)
Comment (linbinskn) [reply pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py:66]: Have rebased. Also add the following rewrite pattern for gpu kernel implementation. The equality between original graph and rewritten graph is not strict. 
    #  weight -> fake-quant  \
    #  input  -> fake-quant -> linear -> fake-quant -> output
    #                   bias /
    #
    # rewrited graph:
    #
    #  weight -> quant  \
    #  input  -> quant -> qgemm -> dequant -> output
    #  bias   -> quant  /

**Review by wyzero — DISMISSED** [review]

**Review by wyzero — DISMISSED** [review]

File: pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py (hunk 1 above)
Comment (chenbohua3) [reply pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py:66]: remove this codes?

Comment (linbinskn) [reply pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py]: done

File: pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+
```
Comment (chenbohua3) [inline pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py:1]: remove these empty lines

File: pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py [hunk 3]
```diff
@@ -294,10 +488,9 @@ def forward(self, x):
...
   }
 }
-
         """
         self._test_torchscipte_to_mhlo(traced_model._c, expect_str, self.pdll_files, enable_int8=True)
 
 
 if __name__ == "__main__":
-    unittest.main()
+    unittest.main()
```
Comment (chenbohua3) [inline pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py:496]: Although we do not enable python style check in this repo, it is better for each developer to obey the pep8 rules. There should be an empty line in the end of each python file

File: pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py (hunk 2 above)
Comment (linbinskn) [reply pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py]: Done.

File: pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py (hunk 3 above)
Comment (linbinskn) [reply pytorch_blade/tests/disc/pdl/test_torchscipt_to_mhlo/test_quantization.py]: Agree with you, have modified.

**Review by chenbohua3 — APPROVED** [review]

**Pull request merged**